components/script/dom/path2d.rs

```diff
@@ -59,20 +63,198 @@ impl Path2D {
     pub(crate) fn segments(&self) -> Vec<PathSegment> {
         self.path.borrow().clone()
     }
+
+    fn is_path_empty(&self) -> bool {
+        self.path.borrow().is_empty()
+    }
+
+    fn last_point(path: &[PathSegment]) -> Option<PathSegment> {
```

Comment:
for this function, since I'm not working with points but `PathSegment` is it ok to just return `PathSegment::ClosePath` if a last point can't be determined? For example, there are only `ClosePath` segments found in the path.

Reply:
Shouldn't this function return a Option<Point2D>? And for all closepaths it should return None (as per spec if subpath is empty closepath should be nop, so we should not even insert it).

Reply:
Did you mean this part?
```rust
let apply_matrix_transform = |x: f32, y: f32| -> Point2D<f64> {
    matrix.transform_point(Point2D::new(x.into(), y.into()))
};
```
If so, after you made the changes to `dommatrix2dinit_to_matrix` function the `Option` wasn't necessary anymore.

Reply:
Oh I didn't return a Point2d because we don't store a list of points but `PathSegment` so I just returned the appropriate `PathSegment` or None.

Reply:
I'm hitting some difficulties with `PathSegment::ArcTo`. From the spec `ArcTo` should use the previous point as its starting point and if there isn't a previous point then it uses the first control point as the starting. So in the situation where a path looks like `[ArcTo, ArcTo, ArcTo]` and I want to get the last point for such a subpath, do I have compute the point generated by the 2nd `ArcTo` in that array in order to use it as the previous point for the last `ArcTo`?

Reply:
Unfortunately, yes.